internal/convert_test.go

```diff
@@ -74,6 +74,43 @@ func TestGetDDL(t *testing.T) {
 	assert.ElementsMatch(t, normalize(e), normalize(ddl))
 }
 
+func TestGetFK(t *testing.T) {
+	conv := MakeConv()
+	conv.SpSchema["table1"] = ddl.CreateTable{
+		Name:     "table1",
+		ColNames: []string{"a", "b"},
+		ColDefs: map[string]ddl.ColumnDef{
+			"a": ddl.ColumnDef{Name: "a", T: ddl.Type{Name: ddl.Int64}},
+			"b": ddl.ColumnDef{Name: "b", T: ddl.Type{Name: ddl.Int64}},
+		},
+		Pks: []ddl.IndexKey{ddl.IndexKey{Col: "a"}},
+		Fks: []ddl.Foreignkey{ddl.Foreignkey{Name: "fk1", Columns: []string{"b"}, ReferTable: "ref_table1", ReferColumns: []string{"ref_c"}}},
+	}
+	conv.SpSchema["table2"] = ddl.CreateTable{
+		Name:     "table2",
+		ColNames: []string{"a", "b", "c"},
+		ColDefs: map[string]ddl.ColumnDef{
+			"a": ddl.ColumnDef{Name: "a", T: ddl.Type{Name: ddl.Int64}},
+			"b": ddl.ColumnDef{Name: "b", T: ddl.Type{Name: ddl.Int64}},
+			"c": ddl.ColumnDef{Name: "c", T: ddl.Type{Name: ddl.Int64}},
+		},
+		Pks: []ddl.IndexKey{ddl.IndexKey{Col: "a"}},
+		Fks: []ddl.Foreignkey{ddl.Foreignkey{Name: "fk2", Columns: []string{"b", "c"}, ReferTable: "ref_table2", ReferColumns: []string{"ref_b", "ref_c"}}},
+	}
+	ddl := conv.GetFK(ddl.Config{ForeignKeys: true})
+	normalize := func(l []string) (nl []string) {
+		for _, s := range l {
+			nl = append(nl, strings.Join(strings.Fields(s), " "))
+		}
+		return nl
+	}
```

Comment:
Instead of normalizing whitespaces in the test, should we just fix the PrintForeignKeyAlterTable function?

Reply:
Yes, agreed.
The issue is that we add an extra space between ADD and FOREIGN KEY.
e.g. we print
```
ALTER TABLE table1 ADD CONSTRAINT fk1  FOREIGN KEY ...
```

Reply:
updated ALTER TABLE statement

Reply:
I see that we are still using normalize function for ASSERT statements. We should fix the PrintCreateTable and/or the unit test to remove the need for the same.

Reply:
removed normalize function and updated expected statements in the tests